packages/wp-story-editor/src/components/metaBoxes/useSaveMetaBoxes.js

```diff
@@ -17,45 +17,47 @@
 /**
  * External dependencies
  */
-import { useEffect, useState, usePrevious } from '@web-stories-wp/react';
+import { useEffect, usePrevious } from '@web-stories-wp/react';
+import { useConfig } from '@web-stories-wp/story-editor';
 
 /**
  * Internal dependencies
  */
-import useMetaBoxes from '../../../integrations/wordpress/metaBoxes/useMetaBoxes';
-import { useAPI } from '../../api';
+import { saveMetaBoxes } from '../../api/metaboxes';
+import useMetaBoxes from './useMetaBoxes';
 
 /**
  * Effect to save meta boxes for a story.
  *
  * @see https://github.com/WordPress/gutenberg/blob/148e2b28d4cdd4465c4fe68d97fcee154a6b209a/packages/edit-post/src/store/effects.js#L24-L126
  * @param {Object} props Hook props.
  * @param {Object} props.story Story object.
- * @param {boolean} props.isSaving Whether saving is in progress.
- * @param {boolean} props.isAutoSaving Whether autosaving is in progress.
+ * @param {boolean} props.isSavingStory Whether saving is in progress.
+ * @param {boolean} props.isAutoSavingStory Whether autosaving is in progress.
  * @return {{isSavingMetaBoxes: boolean}} Metaboxes status.
  */
-function useSaveMetaBoxes({ story, isSaving, isAutoSaving }) {
-  const { hasMetaBoxes, locations } = useMetaBoxes(({ state }) => ({
-    hasMetaBoxes: state.hasMetaBoxes,
-    locations: state.locations,
-  }));
+function useSaveMetaBoxes({ story, isSavingStory, isAutoSavingStory }) {
+  const { hasMetaBoxes, locations, isSavingMetaBoxes, setIsSavingMetaBoxes } =
```

Comment:
Is it weird that this hook re-exports `isSavingMetaBoxes`?

Reply:
Ah, that totally slipped my mind, thank you! That also led to some code simplification. Done in https://github.com/google/web-stories-wp/pull/9027/commits/ea11f5f953ffb7de53e29c6bdeaa374926d775c6